corehq/apps/couch_sql_migration/couchsqlmigration.py

```diff
@@ -236,15 +236,28 @@ def _migrate_form_and_associated_models_async(self, wrapped_form):
             self.processed_docs += 1
             self._log_main_forms_processed_count(throttled=True)
 
-    def _migrate_form_and_associated_models(self, couch_form):
-        sql_form = _migrate_form(self.domain, couch_form)
+    def _migrate_form_and_associated_models(self, couch_form, form_is_processed=True):
+        """
+        Copies `couch_form` into a new sql form
+        """
+        if form_is_processed:
+            with force_phone_timezones_should_be_processed():
+                adjust_datetimes(couch_form.form)
+
+        sql_form = XFormInstanceSQL(
+            form_id=couch_form.form_id,
+            domain=self.domain,
+            xmlns=couch_form.xmlns,
+            user_id=couch_form.user_id,
+        )
```

Comment:
You'll see this is taken from `_migrate_unprocessed_form()`. For processed forms, the original code ends up instantiating XFormInstanceSQL with
```python
        form_id = extract_meta_instance_id(form_data) or six.text_type(uuid.uuid4())
        return XFormInstanceSQL(
            form_id=form_id,
            xmlns=form_data.get('@xmlns'),
            received_on=datetime.datetime.utcnow(),
            user_id=extract_meta_user_id(form_data),
        )
```
and then later changing `None` values for `xmlns` to `""` and overwriting `received_on`. But considering in our case the forms are already processed, we would already have form_id, an xmlns, and user_id. (Right?) So it seemed this was simpler. ... Or is it possible that couch_form doesn't have form_id, or xmlns, or user_id, but couch_form.form does?

Reply:
nvm. Failing tests suggest I should follow original behaviour more closely.

Reply:
Ah, I get it now: "processed" doesn't mean "turned into something reasonable". It means "We've seen this one before. It could be complete rubbish."